frontend/src/components/pages/teacher/ClassroomsPage.tsx

```diff
@@ -1,8 +1,220 @@
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "6",
+      name: "David 6",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "7",
+      name: "David 7",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "8",
+      name: "David 8",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "9",
+      name: "David 9",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+    {
+      id: "10",
+      name: "David 10",
+      studentCount: 200,
+      assessmentCount: 200,
+      grade: Grade.GRADE_2,
+      activeAssessments: 200,
+    },
+  ];
+
+  const {
+    paginatedData,
+    totalPages,
+    currentPage,
+    setCurrentPage,
+  } = usePaginatedData(classrooms, 8);
+
+  const handleTabChange = (index: TabEnumClassroom) => {
+    setTabIndex(index);
+  };
+
+  const handleAddClassroom = () => {
+    setIsModalOpen(true);
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <Box>
+        <HeaderWithButton
+          buttonText="Add New Classroom"
+          onClick={handleAddClassroom}
+          showButton={classrooms.length !== 0}
+          title="Classroom"
+        />
+        <AddClassroomModal
+          isOpen={isModalOpen}
+          onClose={() => setIsModalOpen(false)}
+        />
+      </Box>
+      <Box flex="1">
+        {classrooms.length !== 0 ? (
+          <>
+            <Tabs index={tabIndex} marginTop={3} onChange={handleTabChange}>
+              <TabList>
+                <Tab color={unselectedTabColor}>Active</Tab>
+                <Tab color={unselectedTabColor}>Archived</Tab>
+              </TabList>
+              <TabPanels>
+                <TabPanel padding="0">
+                  <Grid gap={4} templateColumns="repeat(4, 1fr)">
+                    {paginatedData.map((classroom) => (
+                      <GridItem key={classroom.id} flex="1" paddingTop="4">
+                        <ClassroomCard
+                          key={classroom.id}
+                          activeAssessments={classroom.activeAssessments}
+                          assessmentCount={classroom.assessmentCount}
+                          grade={classroom.grade}
+                          name={classroom.name}
+                          studentCount={classroom.studentCount}
+                        />
+                      </GridItem>
+                    ))}
+                  </Grid>
+                  <VStack
+                    alignItems="center"
+                    paddingBottom="6"
+                    paddingTop="6"
+                    spacing="6"
+                    width="100%"
+                  >
+                    {classrooms.length > 8 && (
```

Comment:
nit: don't hard-code `8`
```suggestion
                    {totalPages > 1 && (
```